avocado/utils/asset.py

```diff
@@ -351,18 +323,46 @@ def fetch(self):
 
         raise OSError("Failed to fetch %s." % basename)
 
+    def find_asset_file(self):
+        """
+        Search for the asset file in each one of the cache locations
+
+        :return: asset file if exists or None
+        :rtype: str or None
+        """
+        parsed_url = urllib.parse.urlparse(self.name)
+        basename = os.path.basename(parsed_url.path)
+        cache_relative_dir = self._get_relative_dir(parsed_url)
```

Comment:
`parsed_url`, `basename` and `cache_relative_dir` seem like good candidates for class attributes (with better naming), since we are redefining them the same way in multiple methods.

Reply:
not your commit's fault though, so it's ok to overlook that.

Reply:
I noticed it when I submitted the PR, and my unconscious was telling me that there would be a comment about that :D. I already added this change to the list for the next refactoring round. But thanks anyway for pointing that out.